lib/cisco_node_utils/yum.rb

```diff
@@ -66,9 +76,22 @@ def self.detect_vrf
     end
 
     def self.install(pkg, vrf=nil)
-      vrf = vrf.nil? ? detect_vrf : "vrf #{vrf}"
-      config_set('yum', 'install', pkg, vrf)
+      if platform == :nexus
+        vrf = vrf.nil? ? detect_vrf : "vrf #{vrf}"
+        config_set('yum', 'install', pkg, vrf)
 
+      elsif platform == :ios_xr
+        filename = pkg.strip.tr(':', '/').split('/').last
+        pkg_info = Yum.decompose_name(filename)
+        if pkg_info
+          pkg_name = "#{pkg_info[1]}-#{pkg_info[2]}-#{pkg_info[3]}"
+          debug "Installing package #{pkg_name}"
+          `#{EXEC_IN_DEFAULT_NS} install activate pkg 0x0 #{pkg_name}`
+          `#{EXEC_IN_DEFAULT_NS} install commit sdr`
+        else
+          fail 'Failed to install the requested rpm.'
+        end
+      end
       # HACK: The current nxos host installer is a multi-part command
```

Comment:
Is this check specific to Nexus (in which case please move it up into the if block above), or is it needed for XR as well (in which case please update the comment to make this clear).